### PR TITLE
MGDOBR-119: add BridgeExecutorController

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-119](https://issues.redhat.com/browse/MGDOBR-119) 

This PR aims to add the BridgeExecutor controller: for every Processor that the shard has to deploy, a new Executor deployment + service is going to be created. 

**At the moment the shard does not really deploy the executor application, it simply creates a deployment for whatever `event-bridge.executor.image` is provided**. The PR to configure the executor and deploy it properly will follow https://issues.redhat.com/browse/MGDOBR-122 

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by r00ta** [comment]
Hi @ricardozanini @sutaakar , this should be ready for review. Let me know what do you think

**Review by sutaakar — APPROVED** [review]
Looks good to me

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/providers/TemplateProviderImpl.java [hunk 1]
```diff
@@ -5,48 +5,66 @@
...
 import io.fabric8.kubernetes.api.model.ObjectMeta;
 import io.fabric8.kubernetes.api.model.Service;
 import io.fabric8.kubernetes.api.model.apps.Deployment;
 import io.fabric8.kubernetes.api.model.networking.v1.Ingress;
+import io.fabric8.kubernetes.client.CustomResource;
 import io.fabric8.kubernetes.client.utils.Serialization;
 import io.fabric8.openshift.api.model.Route;
 
 @ApplicationScoped
 public class TemplateProviderImpl implements TemplateProvider {
```
[the remark this answers is not in the document]
Comment (r00ta) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/providers/TemplateProviderImpl.java:20]: https://issues.redhat.com/browse/MGDOBR-157

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/resources/BridgeExecutorStatus.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.redhat.service.bridge.shard.operator.resources;
+
+import java.util.List;
+
+public class BridgeExecutorStatus {
+
+    private List<String> conditions;
+
+    private PhaseType phase;
```
[the remark this answers is not in the document]
Comment (r00ta) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/resources/BridgeExecutorStatus.java:9]: https://issues.redhat.com/browse/MGDOBR-158

File: shard-operator/src/main/resources/templates/bridge-executor-service.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    - apiVersion: ""
+      kind: ""
+      name: ""
+      uid: ""
+spec:
+  ports:
+    - name: web
+      protocol: TCP
+      port: 8080
+      targetPort: 8080
```
[the remark this answers is not in the document]
Comment (r00ta) [reply shard-operator/src/main/resources/templates/bridge-executor-service.yaml:19]: https://issues.redhat.com/browse/MGDOBR-115

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/controllers/BridgeExecutorController.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    @Override
+    public void init(EventSourceManager eventSourceManager) {
+        DeploymentEventSource deploymentEventSource = DeploymentEventSource.createAndRegisterWatch(kubernetesClient, BridgeExecutor.COMPONENT_NAME);
+        eventSourceManager.registerEventSource("bridge-processor-deployment-event-source", deploymentEventSource);
+        ServiceEventSource serviceEventSource = ServiceEventSource.createAndRegisterWatch(kubernetesClient, BridgeExecutor.COMPONENT_NAME);
+        eventSourceManager.registerEventSource("bridge-processor-service-event-source", serviceEventSource);
+    }
+
+    @Override
+    public UpdateControl<BridgeExecutor> createOrUpdateResource(BridgeExecutor bridgeExecutor, Context<BridgeExecutor> context) {
```
[the remark this answers is not in the document]
Comment (r00ta) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/controllers/BridgeExecutorController.java:54]: Done

**Review by ricardozanini — APPROVED** [review]

File: build-parent/pom.xml [hunk 5]
```diff
@@ -27,7 +27,7 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
-    <quarkus.platform.version>2.3.0.Final</quarkus.platform.version>
+    <quarkus.platform.version>2.4.2.Final</quarkus.platform.version>
```
Comment (r00ta) [inline build-parent/pom.xml:30]: Apparently we were hit by https://github.com/quarkusio/quarkus/issues/19494 . After upgrading to 2.4.2, I've got 3 green CI checks on a row. I'd say it fixed my issues with CI @ricardozanini @sutaakar

**Review by r00ta — COMMENTED** [review]
Hi @sutaakar @ricardozanini , I've included a small refactoring of `BridgeExecutor` so that we make use of a Builder (like for `BridgeIngress`). And I've upgraded to quarkus 2.4.2 to fix CI stability.

**Comment by sutaakar** [comment]
Thanks for info

**Pull request merged**